src/dex/swaap-v1/swaap-v1.ts

```diff
@@ -0,0 +1,633 @@
+    return `${this.dexKey}_${poolAddress}`;
+  }
+
+  // Returns list of pool identifiers that can be used
+  // for a given swap. poolIdentifers must be unique
+  // across DEXes. It is recommended to use
+  // ${dexKey}_${poolAddress} as a poolIdentifier
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    return this.findPools(
+      srcToken.address.toLowerCase(),
+      destToken.address.toLowerCase(),
+    ).map(p => this.getPoolIdentifier(p));
+  }
+
+  // Encode params required by the exchange adapter
+  // Used for multiSwap, buy & megaSwap
+  // Hint: abiCoder.encodeParameter() couls be useful
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: SwaapV1Data,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    return {
+      targetExchange: data.pool,
+      payload: '0x',
+      networkFee: '0',
+    };
+  }
+
+  private getSwaapV1Param(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: SwaapV1Data,
+    side: SwapSide,
+  ): [SwaapV1PoolSwapArguments, string] {
+    let functionName;
+    let limitAmount;
+    let amountIn;
+    let amountOut;
+    if (side == SwapSide.SELL) {
+      functionName = 'swapExactAmountInMMM';
+      limitAmount = MAX_INT;
+      amountIn = BigInt(srcAmount);
+      amountOut = BigInt(0);
+    } else {
+      functionName = 'swapExactAmountOutMMM';
+      limitAmount = MAX_INT;
+      amountIn = BigInt(MAX_UINT);
+      amountOut = BigInt(destAmount);
+    }
+
+    const params: SwaapV1PoolSwapArguments = [
+      srcToken,
+      BigInt(amountIn),
+      destToken,
+      BigInt(amountOut),
+      BigInt(MAX_UINT),
+    ];
+
+    return [params, functionName];
+  }
+
+  // Encode call data used by simpleSwap like routers
+  // Used for simpleSwap & simpleBuy
+  // Hint: this.buildSimpleParamWithoutWETHConversion
+  // could be useful
+  async getSimpleParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: SwaapV1Data,
+    side: SwapSide,
+  ): Promise<SimpleExchangeParam> {
+    const [params, functionName] = this.getSwaapV1Param(
+      srcToken,
+      destToken,
+      srcAmount,
+      destAmount,
+      data,
+      side,
+    );
+    const swapData = SwaapV1.poolInterface.encodeFunctionData(
+      functionName,
+      params,
+    );
+    return this.buildSimpleParamWithoutWETHConversion(
+      srcToken,
+      srcAmount,
+      destToken,
+      destAmount,
+      swapData,
+      data.pool,
+    );
+  }
+
+  // // Returns list of top pools based on liquidity. Max
+  // // limit number pools should be returned.
+  async getTopPoolsForToken(
+    tokenAddress: Address,
+    count: number,
+  ): Promise<PoolLiquidity[]> {
+    return this.getTopPoolsForTokens([tokenAddress], count, null).then(
```

Comment:
Here the usage of an http request is ok because we call getTopPoolsForToken only once in a while.